sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -1800,7 +1800,7 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
               if (testCase.insertNullsToStorage) {
                 null
               } else {
-                Row(Seq(Row(3, 4)), Seq(Map(false -> "mno", true -> "pqr")))
+                Row(Seq(Row(1, 2)), Seq(Map(false -> "def", true -> "jkl")))
```

Comment:
```
sql("alter table t alter column s drop default")
sql("insert into t select 2, default")
sql(
          """
            alter table t alter column s
            set default struct(
              array(
                struct(3, 4)),
              array(
                map(false, 'mno', true, 'pqr')))""")
...

```
According to the test case show above, it seems `Row(Seq(Row(3, 4)), Seq(Map(false -> "mno", true -> "pqr")))` is correct.

Reply:
Hi Jiaan, good question:

* It seems actually the correct result for this row should be `2, NULL` :) Because we run `alter table t alter column s drop default` and then `insert into t select 2, default`, the default value should be NULL here.
* This is likely happening because the INSERT operations to the Orc table are all writing to the same column in the same file. The implementation of column default scanning for the Orc data source substitutes the existence default value (assigned when the column was created) when the *entire column* is missing.
* This is a known existing bug that @gengliangwang and I recently discovered. The solution to fix the bug is to switch to a new file after changing the default value in this way. I will prepare a PR to fix this bug separately.

Reply:
Yes. At first time, this row should be `2, NULL`. But after `alter column s set default`, it seems should use the new default value.

Reply:
Note, I fixed this bug for the Orc reader in this PR.